### ISSUE TITLE
Enhancement - Alternative Crafting

### ISSUE THREAD

**Issue by davetgc** [Dark-Basic-Software-Limited/GameGuruRepo#4639]
Add the ability to craftable objects to define craft table, e.g. Workbench, Fire Pit, Ammo Bench, etc. For example, it's unrealistic for a player to be able to make soup at a workbench. 

This would bring the system in line with modern ga

**Comment by Necrym59** [comment]
I agree to a certain extent, crafting needs a catalyst, like a crafting station/object most games do this even if its not an obvious crafting table eg: a carried crafting kit. Perhaps some smaller crafts could be implemented such as making arrows or simple potions by picking up a crafting kit might make sense which could activate the crafting screen at anytime, (This i could implement pretty quick actually). As for the recipes ( i did ask for them to be in a separate recipe container so they were learned recipies, rather than inventory, maybe later) but can be placed at/near a station or traders, some may have to be discovered or part of a quest, that was the whole point of them being separate recipies, a lot of newer games are taking this approach now it also allows for future use in creating crafting skills from them and enhancing the players crafting kit range and capabilities.

**Comment by Necrym59** [comment]
To begin with a crafting kit is now a behavior, other features pertaining to the types of crafting avialable at crafting stations or in the crafting kit will come later
